src/components/Nav/Nav.tsx

```diff
@@ -60,47 +72,71 @@ export class Nav extends React.Component<INavProps, INavState> implements INav {
     return this._selectedKey;
   }
 
-  private _renderLink(link: INavLink, linkIndex: number, nestingLevel: number, hasGroupButton: boolean): React.ReactElement<{}> {
+  private _renderAnchorLink(link: INavLink, linkIndex: number, nestingLevel: number): React.ReactElement<{}> {
     let { onLinkClick } = this.props;
 
     // Determine the appropriate padding to add before this link.
     // In RTL, the "before" padding will go on the right instead of the left.
     const isRtl: boolean = getRTL();
-    const paddingBefore: string = (_indentationSize * nestingLevel + (hasGroupButton ? 40 : 20)).toString(10) + 'px';
-
-    const isLinkSelected: boolean = _isLinkSelected(link, this._selectedKey);
-    if (isLinkSelected) {
-      this._selectedKey = link.key ? link.key : undefined;
-    }
+    const paddingBefore: string = (_indentationSize * nestingLevel +
+      (this._hasExpandButton ? _indentWithExpandButton : _indentNoExpandButton)).toString(10) + 'px';
 
     return (
-      <li role='listitem' key={ linkIndex }>
         <a
-          className={ css('ms-Nav-link', { 'is-selected' : isLinkSelected }) }
+          className={ css('ms-Nav-link', nestingLevel > 0 ? 'ms-Nav-SublinkTextSize' : '') }
           style={ { [isRtl ? 'paddingRight' : 'paddingLeft'] : paddingBefore } }
           href={ link.url || 'javascript:' }
-          onClick={ onLinkClick }
+          onclick={ !!link.onClick ? link.onClick : onLinkClick }
           aria-label={ link.ariaLabel }
-          title={ link.title ? link.title : '' }
-          target={ link.target || '' }
+          title={ link.title ? link.title : link.name }
+          target={ link.target }
         >
-          { (link.iconClassName ?
+          { link.iconClassName ?
```

Comment:
Can be simplified to
{ link.iconClassName && <i className={ css('ms-Icon', 'ms-Nav-IconLink', link.iconClassName) }></i>}